docs/vscode.md

```diff
@@ -3,50 +3,106 @@
+    },
+    "python.testing.pytestArgs": ["flask"],
+    "python.testing.unittestEnabled": false,
+    "python.testing.nosetestsEnabled": false,
+    "python.testing.pytestEnabled": true,
+    "python.formatting.provider": "black",
+    "python.formatting.blackPath": "flask/venv/bin/black",
+    "python.linting.pylintEnabled": true,
+    "python.linting.enabled": true
 }
 ```
+
+## VSCode debugger for flask container
+
+VSCode debugger: [https://code.visualstudio.com/docs/editor/debugging]
+VSCode & remote containers: [https://code.visualstudio.com/docs/remote/containers]
+Using profiels with Compose: [docs.docker.com/compose/profiles]
+
+To set up a development environment with Flask, Docker and VSCode when the docker context points to the local machine, see the steps below. If the Flask application container is running on the VM, an additional step is required to set the local docker context to point to the docker instance on the VM.
+
+To directly enter a Docker development that supports debugging in the container and launch the debugger:
+
+-   Step 1: `docker-compose up -d mysql minio app_debug`
+-   Step 2, Click on remote connection icon in VSCode, choose "Attach to running container".
+-   Step 3, When the prompt "Select the container to the attach VS Code" comes up, choose "/stager-app_debug_1".
+-   Step 4, In the VSCode window that runs in the app container, use F5 to launch the debugger.
+
+To start the Docker development without debugging functionality enabled, and later enable the debugger functionality:
+
+-   Step 1: `docker-compose up`
+-   Step 2: Stop and remove the Docker container `app`
+-   Step 3: `docker-compose up app_debug`
+-   Step 4, Click on remote connection icon in VSCode, choose "Attach to running container".
+-   Step 5, When the prompt "Select the container to the attach VS Code" comes up, choose "/stager-app_debug_1".
+-   Step 6, In the VSCode window that runs in the app container, use F5 to launch the debugger.
+
+Explanation:
+
+-   Debugpy extension is already added to `requirements-dev.txt` and `requirements-dev.in`.
+-   `app_debug` is a service added in `docker-compose.yaml`. By using profile, this service is only started when it is targeted.
+
+# Other explored options:
+
+### Option 1: Reopen the VSCode in container, starting the flask app only when the debugger session starts in the container.
+
+Conor's working solution.
+Set the entrypoint to `tail -f /dev/null` in `./devcontainer/docker-compose.yml` such that the flask app is not started when we reopen the VSCode session in the container. When we launch the debugger, the app is started using the `launch` request in `.vscode/launch.json`.
+
+Install the necessary extensions for the container in `.vscode/extensions.json`.
+
+### Option 2: Attach to a running container, kill the flask app right before the debugger is launched.
+
+This is an ideal solution. We don't have a working version of it yet. Here's the proposed workflow:
```

Comment:
Is this an 'ideal' solution if it doesn't exist?

Reply:
I will remove all the options because they are still in the experimental stage (except that option 1 is Conor's working solution. However option 1 is not about attaching to a running container, so I think it's ok to not include it). Also, these options will require more modifications to the settings in `docker-compose` files or `.vscode` directory than the current solution using "profile".